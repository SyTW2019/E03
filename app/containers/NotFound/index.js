import React from 'react';
import { Link } from 'react-router-dom';
import PageNotFound from './images/PageNotFound.jpg';

export default function NotFound() {
  
  return(
    <div>
    <img src={PageNotFound} style={{display: 'block', margin: 'auto', position: 'relative' }} alt="Error, pÃ¡gina no encontrada" />
    </div>
  );
}
