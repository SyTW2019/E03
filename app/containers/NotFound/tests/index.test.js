import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from 'react-testing-library';
import "jest-dom/extend-expect";
import renderer from "react-test-renderer";

import NotFound from '../index';

const src = 'test.png';
const alt = 'test';
const renderComponent = (props = {}) =>
  render(<NotFound src={src} alt={alt} {...props} />);


describe('<NotFound />', () => {

    afterEach(cleanup);

    it('render', () => {
        const div = document.createElement(div);
        ReactDOM.render(<NotFound/>,div);
      });

      it('should render an <img> tag', () => {
        const { container } = renderComponent();
        const element = container.querySelector('img');
        expect(element).not.toBeNull();
      });

      it('should have an src attribute', () => {
        const { container } = renderComponent();
        const element = container.querySelector('img');
        expect(element.hasAttribute('src')).toBe(true);
      });

      it('should have an alt attribute', () => {
        const { container } = renderComponent();
        const element = container.querySelector('img');
        expect(element.hasAttribute('alt')).toBe(true);
      });

      it('should not have a class attribute', () => {
        const { container } = renderComponent();
        const element = container.querySelector('img');
        expect(element.hasAttribute('class')).toBe(false);
      });
    
      it('should not adopt a srcset attribute', () => {
        const srcset = 'test-HD.png 2x';
        const { container } = renderComponent({ srcset });
        const element = container.querySelector('img');
        expect(element.hasAttribute('srcset')).toBe(false);
      });

      it('Matches snapshots', () => {
        const tree = renderer.create(<NotFound title="save"/>).toJSON();
        expect(tree).toMatchSnapshot();
      });

});