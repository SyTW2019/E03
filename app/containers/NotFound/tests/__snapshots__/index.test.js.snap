// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Matches snapshots 1`] = `
<div>
  <img
    alt="Error, pÃ¡gina no encontrada"
    src="IMAGE_MOCK"
    style={
      Object {
        "display": "block",
        "margin": "auto",
        "position": "relative",
      }
    }
  />
</div>
`;
